projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
 })
 
 export class IgxButtonGroupComponent implements AfterViewInit {
+    /**
+     * @hidden
+     */
     @ViewChildren(IgxButtonDirective) public buttons: QueryList<IgxButtonGroupComponent>;
 
+    /**
+     * An @Input property that sets the value of the `id` attribute.
+     * ```html
+     *  <igx-buttongroup [id]="'igx-dialog-56'" [multiSelection]="!multi" [values]="alignOptions">
+     * ```
+     */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-buttongroup-${NEXT_ID++}`;
 
+    /**
+     * Allows you to set a style using the `itemContentCssClass` input.
+     *```typescript
+     *public style1 = "styleClass";
+     * //..
+     *```
+     * ```html
+     *<igx-buttongroup [itemContentCssClass]="style1" [multiSelection]="!multi" [values]="alignOptions">
+     *```
+     */
     @Input() set itemContentCssClass(value: string) {
         this._itemContentCssClass = value || this._itemContentCssClass;
     }
+
+    /**
+     * Returns the cssClass of the `igx-buttongroup`.
```

Comment:
@alexalex972 "Returns the CSS class of the item content of the `IgxButtonGroup`"